test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,224 @@
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function getMetadata(): array
+                        {
+                            return [];
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function tag($tags): ItemInterface
+                        {
+                            Assert::assertSame(SpecificationLoader::CACHE_TAG, $tags);
+
+                            return $this;
+                        }
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        public function expiresAt($expiration)
+                        {
+                            return $this;
+                        }
+                    });
+                    $this->specs[$key] = $item;
+
+                    return $item;
+                }
+
+                /**
+                 * {@inheritDoc}
+                 */
+                public function invalidateTags(array $tags)
+                {
+                    return true;
+                }
+
+                public function delete(string $key): bool
+                {
+                    return true;
+                }
+            }
+        );
+
+        parent::setUp();
+    }
+
+    public function testRegisterSpecAndGettingRegisteredSpecification(): void
+    {
+        $specificationFileName       = 'specification.json';
+        $specificationArray          = $this->getSpecificationArray($specificationFileName);
+        $expectedSpecificationConfig = new SpecificationConfig(
+            $specificationArray['path'],
+            $specificationArray['type'] ?? null,
+            $specificationArray['name_space'],
+            $specificationArray['media_type']
+        );
+
+        $expectedSpecifications = [self::SPECIFICATION_NAME => $expectedSpecificationConfig];
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specifications = $this->specificationLoader->list();
+
+        Assert::assertEquals($expectedSpecifications, $specifications);
+
+        $specificationConfig = $this->specificationLoader->get(self::SPECIFICATION_NAME);
+        Assert::assertEquals($expectedSpecificationConfig, $specificationConfig);
+    }
+
+    public function testLoadYamlFromCache(): void
+    {
+        $specificationFileName = 'specification.yaml';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specification = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+
+        rename(__DIR__ . '/' . $specificationFileName, __DIR__ . '/' . $specificationFileName . '_test');
```

Comment:
When you implement the in-memory cache mentioned above, you will have full control of the cache contents without the need to touch the file system.
First set up the cache class, put a cached spec there, then call load, verify that parseOpenApi was not called on a mock of SpecificationParser and locate was not called on a mock of FileLocatorInterface.